src/main/java/com/java401d18/songr/controllers/AlbumController.java

```diff
@@ -1,58 +1,83 @@
 package com.java401d18.songr.controllers;
 
 import com.java401d18.songr.model.Album;
+import com.java401d18.songr.model.Song;
 import com.java401d18.songr.repositories.AlbumRepository;
+import com.java401d18.songr.repositories.SongRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.servlet.view.RedirectView;
+
 import java.util.List;
 
 @Controller
 public class AlbumController {
-    //Step6: add an Autowired instance of repository to the controller
-    @Autowired
-    AlbumRepository albumRepository;
 
-    @GetMapping("/albums")
-    public String albums(Model model) {
-        List<Album> albumList = albumRepository.findAll();
-        model.addAttribute("albumList", albumList);
-        return "albums";
-    }
+    private final AlbumRepository albumRepository;
+    private final SongRepository songRepository;
 
 
-    @GetMapping("/add-album")
-    public String getAlbumsPage() {
-        return "add-album.html";
-    }
-
-    @PostMapping("/add-album")
-    public RedirectView addAlbumFromForm(String title, String artist, int songCount, int length, String imageUrl) {
-        Album newAlbum = new Album(title, artist, songCount, length, imageUrl);
-        albumRepository.save(newAlbum);
-        return new RedirectView("/albums");
+    @Autowired
+    public AlbumController(AlbumRepository albumRepository, SongRepository songRepository) {
+        this.albumRepository = albumRepository;
+        this.songRepository = songRepository;
     }
-}
-
-
 
 
+    @GetMapping("/albums")
+    public String albums(Model model) {
+        List<Album> albums = albumRepository.findAll();
+        model.addAttribute("albums", albums);
+        return "albums.html";
+    }
 
+    @GetMapping("/addAlbum")
+    public String addAlbumForm(Model model) {
+        model.addAttribute("album", new Album());
+        return "addAlbum.html";
+    }
 
 
 
+    @PostMapping("/addAlbum")
+    public String addAlbumSubmit(@ModelAttribute Album album) {
+        albumRepository.save(album);
+        return "redirect:/albums";
+    }
 
 
+    //lab 13
+    @GetMapping("/songs")
+    public String songs(Model model) {
+        List<Song> songs = songRepository.findAll();
+        model.addAttribute("songs", songs);
```

Comment:
Make sure you're also passing in an "albums" attribute here - otherwise, the albums drop-down in your add songs form won't work.